packages/@guidesmiths/cuckoojs-schematics/src/install-packages/install-packages.factory.ts

```diff
@@ -0,0 +1,55 @@
+import {
+	type Rule,
+	type SchematicContext,
+	type Tree,
+} from '@angular-devkit/schematics';
+import {normalize} from '@angular-devkit/core';
+import {execSync} from 'child_process';
+import {NodePackageInstallTask} from '@angular-devkit/schematics/tasks'
+
+interface Options {
+	directory: string;
+}
+
+type PackageManager = 'npm' | 'pnmp' | 'yarn';
+
+export function main(options: Options): Rule {
+	return function (tree: Tree, context: SchematicContext) {
+		context.logger.info('Installing packages...');
+    const packageManager = getPackageManager(tree, options.directory);
+		return installDependencies(packageManager)
+	};
+}
+
+// inspired by https://github.com/egoist/detect-package-manager
+function getFromLockFile(tree: Tree, directory: string): PackageManager | undefined {
+  if(tree.exists(normalize(`${directory}/package-json.lock`))) return 'npm';
+  if(tree.exists(normalize(`${directory}/yarn.lock`))) return 'yarn';
+  if(tree.exists(normalize(`${directory}/pnpm-lock.yaml`))) return 'pnmp';
+}
+
+// inspired by https://github.com/egoist/detect-package-manager
+function getFromGlobalInstallation(): PackageManager {
+  try {
+    execSync('yarn --version', {stdio: 'pipe'});
+    return 'yarn'
+  } catch(e) {
+    try {
+      execSync('pnpm --version', {stdio: 'pipe'});
+      return 'pnmp';
+    } catch(e) {
+      return 'npm';
```

Comment:
I understand that by default it will use npm if it doesn't find anything. What happens if npm is not installed?

Reply:
And one more question. What happens if several of them are installed? We're giving preference to `yarn`.

Reply:
Preference is based on the lock file. If there is no lock file, it will explore the globally installed ones. In that case, `yarn` takes preference indeed. We can change it.

If the package manager is not installed, it will fail.

Reply:
I don't mind the order. I was just wondering if it's useful to have an optional config parameter to set the preference. Or perhaps I'm suggesting nonsense! 😬

Reply:
Do you mean to overwrite the _discovered_ package manager?